config/config.go

```diff
@@ -263,6 +262,167 @@ func FromViper(v *viper.Viper) (*Config, error) {
 	return cfg, nil
 }
 
+func determineTreeRoot(v *viper.Viper, cfg *Config, logger *log.Logger) error {
+	var err error
+
+	// enforce the various tree root options are mutually exclusive
+	// some of this is being done for us at the flag level, but you can also set these values in config or environment
+	// variables.
+	count := 0
+
+	if cfg.TreeRoot != "" {
+		count++
+	}
+
+	if cfg.TreeRootCmd != "" {
+		count++
+	}
+
+	if cfg.TreeRootFile != "" {
+		count++
+	}
+
+	if count > 1 {
+		return errors.New("only one of tree-root, tree-root-cmd or tree-root-file can be specified")
+	}
+
+	// set git-based tree root command if the walker is git and no tree root has been specified
+	if cfg.Walk == walk.Git.String() && count == 0 {
+		cfg.TreeRootCmd = "git rev-parse --show-toplevel"
+
+		logger.Infof(
+			"git walker enabled and tree root has not been specified: defaulting tree-root-cmd to '%s'",
+			cfg.TreeRootCmd,
+		)
+	}
```

Comment:
This new behaviour should probably be documented too.

I believe the docs currently state `tree-root`:

> Defaults to the directory containing the config file.

Now that should say something like:

> Defaults to the root of the current git worktree. If not in a git repo, defaults the directory containing the config file.

Reply:
I've updated the description.